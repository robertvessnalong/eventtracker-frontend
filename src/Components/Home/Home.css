img {
  width: 100%;
}

.favorite-btn {
  font-size: 32px;
  cursor: pointer;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.center-content {
  margin: auto;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.Event-Items-Home {
  position: relative;
}

.Sign-Up-Btn {
  display: none !important;
  position: absolute;
  z-index: 2;
  right: 0;
  bottom: 0;
  left: 0;
  height: 100%;
  background: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiA/Pgo8c3ZnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgd2lkdGg9IjEwMCUiIGhlaWdodD0iMTAwJSIgdmlld0JveD0iMCAwIDEgMSIgcHJlc2VydmVBc3BlY3RSYXRpbz0ibm9uZSI+CiAgPGxpbmVhckdyYWRpZW50IGlkPSJncmFkLXVjZ2ctZ2VuZXJhdGVkIiBncmFkaWVudFVuaXRzPSJ1c2VyU3BhY2VPblVzZSIgeDE9IjAlIiB5MT0iMCUiIHgyPSIwJSIgeTI9IjEwMCUiPgogICAgPHN0b3Agb2Zmc2V0PSIwJSIgc3RvcC1jb2xvcj0iI2ZmZmZmZiIgc3RvcC1vcGFjaXR5PSIwIi8+CiAgICA8c3RvcCBvZmZzZXQ9IjcwJSIgc3RvcC1jb2xvcj0iI2ZmZmZmZiIgc3RvcC1vcGFjaXR5PSIxIi8+CiAgPC9saW5lYXJHcmFkaWVudD4KICA8cmVjdCB4PSIwIiB5PSIwIiB3aWR0aD0iMSIgaGVpZ2h0PSIxIiBmaWxsPSJ1cmwoI2dyYWQtdWNnZy1nZW5lcmF0ZWQpIiAvPgo8L3N2Zz4=);
  background: -moz-linear-gradient(
    top,
    rgba(255, 255, 255, 0) 0%,
    rgba(255, 255, 255, 1) 70%
  );
  background: -webkit-gradient(
    linear,
    left top,
    left bottom,
    color-stop(0%, rgba(255, 255, 255, 0)),
    color-stop(70%, rgba(255, 255, 255, 1))
  );
  background: -webkit-linear-gradient(
    top,
    rgba(255, 255, 255, 0) 0%,
    rgba(255, 255, 255, 1) 70%
  );
  background: -o-linear-gradient(
    top,
    rgba(255, 255, 255, 0) 0%,
    rgba(255, 255, 255, 1) 70%
  );
  background: -ms-linear-gradient(
    top,
    rgba(255, 255, 255, 0) 0%,
    rgba(255, 255, 255, 1) 70%
  );
  background: linear-gradient(
    to bottom,
    rgba(255, 255, 255, 0) 0%,
    rgba(255, 255, 255, 1) 70%
  );
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#00ffffff', endColorstr='#ffffff',GradientType=0 );
}

.Comment {
  padding: 25px 10px;
  display: none;
}

.Active {
  display: block !important;
}

.Active-Flex {
  display: flex !important;
  justify-content: center !important;
  align-items: center !important;
  text-align: center;
}

.Comment button {
  padding: 5px;
  margin: 10px 0px;
}

svg {
  pointer-events: none;
}
