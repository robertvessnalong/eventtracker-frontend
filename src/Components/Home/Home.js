import {
  Form,
  Row,
  FloatingLabel,
  Col,
  Button,
  Container,
  Alert,
} from 'react-bootstrap';
import { useState, useContext } from 'react';
import { Link } from 'react-router-dom';
import './Home.css';
import EventsFinderApi from '../../API/api';
import EventItem from '../Events/EventItem';
import uuid from 'react-uuid';
import UserContext from '../../Context/UserContext';

const Home = () => {
  const formInitalState = {
    search: '',
  };
  const { loggedIn } = useContext(UserContext);
  const [formData, setFormData] = useState(formInitalState);
  const [events, setEvents] = useState([]);
  const [hasEvents, setHasEvents] = useState(false);
  const [error, setError] = useState(false);
  const [errorMsg, setErrorMsg] = useState([]);
  const [fade, setFadeOut] = useState(false);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({
      ...data,
      [name]: value,
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    const res = await EventsFinderApi.getEvents(formData);
    if (res.error) {
      setError(true);
      setErrorMsg(res.error);
    } else {
      if (res.length > 0) {
        setFadeOut(true);
        setHasEvents(false);
        setFormData(formInitalState);
        setEvents(res.slice(0, 2));
      } else {
        setHasEvents(true);
      }
    }
  };

  return (
    <div className='Home Center'>
      <header className='center-content'>
        <h2>Events</h2>
        <span>Find Events Near You</span>
        <hr></hr>
      </header>
      <Container>
        <div className='Search'>
          <Form onSubmit={handleSubmit} className='mt-4'>
            {error && <Alert variant='danger'>{errorMsg}</Alert>}
            <Row className='g-2'>
              <Col md>
                <FloatingLabel
                  controlId='floatingInputGrid'
                  label='Search Events...'
                >
                  <Form.Control
                    type='text'
                    name='search'
                    value={formData.search}
                    onChange={handleChange}
                  />
                </FloatingLabel>
              </Col>
              <Col md={3}>
                <Button
                  type='submit'
                  className='h-100 w-100 blue-btn'
                  variant='primary'
                >
                  Find Events
                </Button>{' '}
              </Col>
            </Row>
          </Form>
        </div>
      </Container>
      <Container className='Event-Items-Home'>
        {hasEvents && (
          <Alert className='mt-4' variant='warning'>
            Sorry, no events found! Try Again
          </Alert>
        )}
        <Row xs={1} xl={2}>
          {events.map((event) => (
            <Col key={uuid()}>
              <EventItem event={event} />
            </Col>
          ))}
        </Row>
        <Row className={fade ? 'Sign-Up-Btn Active-Flex' : 'Sign-Up-Btn'}>
          <Col>
            {!loggedIn ? (
              <Link to='/login' className='blue-btn'>
                Sign Up To See More
              </Link>
            ) : (
              <Link to='/events' className='blue-btn'>
                See More
              </Link>
            )}
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default Home;
