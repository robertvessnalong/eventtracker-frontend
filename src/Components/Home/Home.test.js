import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from './Home';
import TestUser from '../../Testing/TestUser';
import UserContext from '../../Context/UserContext';
import { BrowserRouter } from 'react-router-dom';

describe('Home Component', () => {
  // Smoke Test
  test('renders without crashing', async () => {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <Home />
        </UserContext.Provider>
      </BrowserRouter>
    );

    const headerSpan = screen.getByText(/Find Events Near You/i);
    expect(headerSpan).toBeInTheDocument();
  });

  test('fireSubmit', async () => {
    const { getByText, getByRole } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <Home />
        </UserContext.Provider>
      </BrowserRouter>
    );
    await waitFor(() => {
      fireEvent.change(getByRole('textbox', { name: /search/i }), {
        query: '',
      });
      fireEvent.submit(getByText('Find Events'));
    });
    const alertMsg = screen.getByText(/Sign Up To See More/i);
    expect(alertMsg).toBeInTheDocument();
  });

  // Snapshot
  test('matches snapshot', async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <Home />
        </UserContext.Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
