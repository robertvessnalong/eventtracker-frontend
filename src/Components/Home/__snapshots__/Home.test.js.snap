// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home Center"
  >
    <header
      class="center-content"
    >
      <h2>
        Events
      </h2>
      <span>
        Find Events Near You
      </span>
      <hr />
    </header>
    <div
      class="container"
    >
      <div
        class="Search"
      >
        <form
          class="mt-4"
        >
          <div
            class="g-2 row"
          >
            <div
              class="col-md"
            >
              <div
                class="form-floating"
              >
                <input
                  class="form-control"
                  id="floatingInputGrid"
                  name="search"
                  type="text"
                  value=""
                />
                <label
                  for="floatingInputGrid"
                >
                  Search Events...
                </label>
              </div>
            </div>
            <div
              class="col-md-3"
            >
              <button
                class="h-100 w-100 blue-btn btn btn-primary"
                type="submit"
              >
                Find Events
              </button>
               
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="Event-Items-Home container"
    >
      <div
        class="row row-cols-xl-2 row-cols-1"
      />
      <div
        class="Sign-Up-Btn row"
      >
        <div
          class="col"
        >
          <a
            class="blue-btn"
            href="/login"
          >
            Sign Up To See More
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
